src/datachain/query/dataset.py

```diff
@@ -352,10 +371,39 @@ def process_udf_outputs(
     udf: "UDFAdapter",
     cb: Callback = DEFAULT_CALLBACK,
     batch_size: int = INSERT_BATCH_SIZE,
+    processed_table: "Table | None" = None,
 ) -> None:
     # Optimization: Compute row types once, rather than for every row.
     udf_col_types = get_col_types(warehouse, udf.output)
 
+    # Track which input sys__ids we've already written to processed table
+    all_processed_sys_ids: set[int] = set()
+
+    def _batch_callback(batch: list[dict[str, Any]]) -> None:
+        """Called after each batch of outputs is inserted.
+
+        Extracts input sys__ids from the actual inserted batch and writes them
+        to the processed table.
+        """
+        if processed_table is None:
+            return
+
+        # Extract sys__ids from ACTUAL inserted rows (tracking_field preserved in
+        # callback)
+        sys_ids = {row["_input_sys_id"] for row in batch if "_input_sys_id" in row}
```

Comment:
I still can not understand the logic with tracking field and this `_input_sys_id`. Wound you mind to explain it?

- Where `_input_sys_id` comes from?
- Why not to use `sys__id`?